float32 x_min
float32 x_max
---